src/main/webapp/app/exercises/programming/manage/programming-exercise-detail.component.ts

```diff
@@ -171,76 +183,105 @@ export class ProgrammingExerciseDetailComponent implements OnInit, OnDestroy {
                     `/exercise-groups/${this.programmingExercise.exerciseGroup?.id}/exercises/${this.programmingExercise.exerciseGroup?.exam?.id}/`;
             }
 
-            this.programmingExerciseService.findWithTemplateAndSolutionParticipationAndLatestResults(programmingExercise.id!).subscribe((updatedProgrammingExercise) => {
-                this.programmingExercise = updatedProgrammingExercise.body!;
-
-                this.setLatestCoveredLineRatio();
-                this.loadingTemplateParticipationResults = false;
-                this.loadingSolutionParticipationResults = false;
-                this.profileService.getProfileInfo().subscribe(async (profileInfo) => {
-                    if (profileInfo) {
-                        if (this.programmingExercise.projectKey && this.programmingExercise.templateParticipation && this.programmingExercise.templateParticipation.buildPlanId) {
-                            this.programmingExercise.templateParticipation.buildPlanUrl = createBuildPlanUrl(
-                                profileInfo.buildPlanURLTemplate,
-                                this.programmingExercise.projectKey,
-                                this.programmingExercise.templateParticipation.buildPlanId,
-                            );
-                        }
-                        if (this.programmingExercise.projectKey && this.programmingExercise.solutionParticipation && this.programmingExercise.solutionParticipation.buildPlanId) {
-                            this.programmingExercise.solutionParticipation.buildPlanUrl = createBuildPlanUrl(
-                                profileInfo.buildPlanURLTemplate,
-                                this.programmingExercise.projectKey,
-                                this.programmingExercise.solutionParticipation.buildPlanId,
-                            );
+            this.templateAndSolutionParticipationSubscription = this.programmingExerciseService
```

Comment:
Only added `this. templateAndSolutionParticipationSubscription = [...]` here, due to the wrap to the next line there is more marked as diff than actually changed